tensorflow/compiler/tf2tensorrt/utils/trt_lru_cache.h

```diff
@@ -123,12 +123,21 @@ struct EngineContext {
   EngineContext(
       TrtUniquePtrType<nvinfer1::ICudaEngine>&& input_cuda_engine,
       TrtUniquePtrType<nvinfer1::IExecutionContext>&& input_execution_context)
+      : cuda_engine(std::move(input_cuda_engine)) {
+    execution_context.push_back(std::move(input_execution_context));
+  }
+  EngineContext(TrtUniquePtrType<nvinfer1::ICudaEngine>&& input_cuda_engine,
+                std::vector<TrtUniquePtrType<nvinfer1::IExecutionContext>>&&
+                    input_execution_context)
       : cuda_engine(std::move(input_cuda_engine)),
         execution_context(std::move(input_execution_context)) {}
 
   mutex mu;
   TrtUniquePtrType<nvinfer1::ICudaEngine> cuda_engine;
-  TrtUniquePtrType<nvinfer1::IExecutionContext> execution_context
+
+  // In explicit batch mode, we maintain a vector of contexts for each engine,
+  // where each context is created for a different profile.
+  std::vector<TrtUniquePtrType<nvinfer1::IExecutionContext>> execution_context
```

Comment:
I understand that this is not new in this PR, but do you know why we guard this field with a mutex?
Somewhere in routine TRTEngineOp::ExecuteTrtEngine, there is this comment "nvinfer1::IExecutionContext::enqueue is not thread safe and we need a mutex".
Can we document the reason for GUARDED_BY(mu) here?